UVa/BasicRemains.java

```diff
@@ -0,0 +1,76 @@
+
+/*
+Given a base b and two non-negative base b integers
+p and m, compute p mod m and print the
+result as a base-b integer. p mod m is defined
+as the smallest non-negative integer k such that
+p = a ∗ m + k for some integer a.
+Input
+Input consists of a number of cases. Each case is
+represented by a line containing three unsigned
+integers. The first, b, is a decimal number between
+2 and 10. The second, p, contains up to 1000 digits between 0 and b − 1. The third, m, contains
+up to 9 digits between 0 and b − 1. The last case is followed by a line containing ‘0’.
+Output
+For each test case, print a line giving p mod m as a base-b integer.
+Sample Input
+2 1100 101
+10 123456789123456789123456789 1000
+0
+Sample Output
+10
+789
+ */
+import static java.lang.Integer.parseInt;
+import static java.lang.System.exit;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.math.BigInteger;
+import java.util.Scanner;
+import java.util.StringTokenizer;
+
+public class BasicRemains {
+
+	static void solve() throws Exception {
+		Scanner input = new Scanner(System.in);
+
+		while (input.hasNext()) {
+			int baseNumber = input.nextInt();
+			if (baseNumber == 0) {
+				break;
+			}
+			BigInteger p = new BigInteger(input.next(), baseNumber);
+			BigInteger m = new BigInteger(input.next(), baseNumber);
+			System.out.println((p.mod(m)).toString(baseNumber));
+		}
+		input.close();
+	}
+
+	static int nextInt() throws IOException {
```

Comment:
I have used a template that a friend sent me. It is a general template that is supposed to be useful and avoid the necessity to repeat the same procedures of getting input and output the result for some common coding challenges that are at Online Judges. This is a similar template with the one that I used
https://github.com/ifhuang/Problems/blob/master/src/codeforces/util/Template7.java

However, in this problem, I had to get a BigInteger as an input, so I used Scanner to get the input from the console, but did not remove the methods of the template that are automatically inserted by the IDE when a new class is created.

I used .close() method of Scanner, because of resource leaks. However, I researched more on this and decided to remove it, as JVM will close it.